src/main/java/ruslan/shastkiv/bookstore/service/book/BookServiceImpl.java

```diff
@@ -56,9 +64,22 @@ public Page<BookDto> search(BookSearchParametersDto searchParametersDto, Pageabl
                 .map(bookMapper::toDto);
     }
 
+    @Override
+    public Page<BookDtoWithoutCategoryIds> getAllBooksByCategoryId(Long id, Pageable pageable) {
+        return bookRepository.findBooksByCategoryId(id, pageable)
+                .map(bookMapper::toDtoWithoutCategoryIds);
+    }
+
     private Book findBookById(Long id) {
         return bookRepository.findById(id).orElseThrow(()
                 -> new EntityNotFoundException("Can t find book by id: " + id));
     }
 
+    private Set<Category> initCategories(List<Long> categoryIds) {
+        return categoryIds.stream()
+                .map(id -> categoryRepository.findById(id).orElseThrow(()
+                        -> new EntityNotFoundException("Cant find category by id: " + id)))
+                .collect(Collectors.toSet());
```

Comment:
What if you want to set 1000 categories to book? You will make a DB request to get each separately this is not efficient. so please think how to such operation using 1 request only

Reply:
I agree with you, but I was also thinking about how to notify the admin which category is missing in the database.